cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -104,6 +106,13 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     }
     boolean useInternalIP = false;
 
+    if (conf.isInNetwork()) {
+      useInternalIP = true;
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(network), "The network to which the instance is peered to " +
+        "must be specified when in network option is selected.");
+      // TODO: If isInNetwork is set to true check if the specified network is peered
+    }
+
     if (network == null && projectId.equals(systemProjectId)) {
```

Comment:
though it is logically the same since there is a check that the network is not null above, I think it's better if this were an 'else if' to make it more clear that these conditions are mutually exclusive, and to make it harder to mess up if the logic is extended in the future.